+sw_tests/+unit_tests/unittest_spinw_spinwave.m

```diff
@@ -0,0 +1,471 @@
+            commensurate_spec = swobj.spinwave(hkl);
+            swobj.genmagstr('mode', 'direct', 'k', [0.123 0 0], 'n', [1 0 0], 'S', [0; 1; 0]);
+            % Forcing incomm struct means exchange parameters don't agree, so we need 'hermit' false
+            incomm_spec = swobj.spinwave(hkl, 'hermit', false);
+            testCase.assertEqual(size(incomm_spec.omega, 1), size(commensurate_spec.omega, 1) * 3);
+        end
+        function test_twin(testCase)
+            % Tests that setting twins gives correct outputs
+            % Create copy to avoid changing obj for other tests
+            swobj_twin = copy(testCase.swobj);
+            swobj_twin.addtwin('axis', [0 0 1], 'phid', [60 120], 'vol', [1 2]);
+            rotc = swobj_twin.twin.rotc;
+            hkl = [1 2; 3 4; 5 6];
+            sw_out = swobj_twin.spinwave(hkl);
+
+            expected_sw = testCase.default_spinwave;
+            expected_sw.param.notwin = false;
+            expected_sw.omega = {};
+            expected_sw.Sab = {};
+            expected_sw.obj = copy(swobj_twin);
+            expected_sw.hkl = hkl;
+            expected_sw.hklA = [2/3 4/3; 0.75, 1; 1.25, 1.5]*pi;
+            % Recalculate without twins for each set of hkl's and compare
+            [~, rotQ] = swobj_twin.twinq([0;0;0]);
+            for ii = 1:3
+                sw_single = testCase.swobj.spinwave((hkl' * rotQ(:, :, ii))');
+                expected_sw.omega = [expected_sw.omega sw_single.omega];
+                rot = rotc(:, :, ii);
+                rot_Sab = zeros(3, 3, 2, 2);
+                % Twin Sab is a rotation of single Sab
+                for jj = 1:2
+                    for kk = 1:2
+                        rot_Sab(:, :, jj, kk) = rot*sw_single.Sab(:, :, jj, kk)*rot';
+                    end
+                end
+                expected_sw.Sab = [expected_sw.Sab rot_Sab];
+            end
+            testCase.verify_spinwave(expected_sw, sw_out);
+        end
+        function test_notwin(testCase)
+            % Create copy to avoid changing obj for other tests
+            swobj_twin = copy(testCase.swobj);
+            swobj_twin.addtwin('axis', [0 0 1], 'phid', [60 120], 'vol', [1 2]);
+            qpts = testCase.qh5;
+            sw_out = swobj_twin.spinwave(qpts, 'notwin', true);
+            % Test even if twin is added to structure it is not actually
+            % calculated if notwin is specified
+            expected_sw = testCase.get_expected_sw_qh5;
+            expected_sw.obj.twin = swobj_twin.twin;
+            testCase.verify_spinwave(expected_sw, sw_out);
+        end
+        function test_cmplxBase_equivalent_with_tri(testCase)
+            % For this structure, both cmplxBase true and false give the
+            % same e-vectors so should give the same result
+            qpts = {[0 0 0], [1 0 0], 5};
+            sw_out = testCase.swobj_tri.spinwave(qpts);
+            sw_out_cmplx = testCase.swobj_tri.spinwave(qpts, 'cmplxBase', true);
```

Comment:
Is there an example where we could validate that the eigenvector elements are different in the two coordinate systems here?

Reply:
Would it be a good idea to pick a system/moment direction for which the vectors would not be the same? For example the basis transformation could be an identity matrix and this test would still pass.

Reply:
But in that case `test_cmplxBase_fails_with_chain` would fail because it wouldn't produce an error for the chain, so we would know. It would be nice to have a system that produces different results and doesn't error out but I don't know of any or really understand how it works... @mducle?

Reply:
To be honest I'm not sure how the `'cmplxBase', true` option is meant to work or what its application really is...

As for testing it, I think we would need to define another model to test its consistency with the default `'cmplxBase', false`, and rather than testing the eigenvectors we would have to test `Sab` or another physical observable.

This is because the Hamiltonian is expressed in a basis defined by the magnon creation/anihilation operators which are in turn quantised along the `e3` direction. If we change this direction we are effectively changing the basis of the Hamiltonian so the eigenvectors we get for `'cmplxBase', true` would be in a different basis than for `'cmplxBase', false` and so would not be numerically the same. But observables such as the spin-spin correlation tensor `Sab` _should_ still be the same (and also the eigenvalues `omega` should also be the same).

As for the model to test it, it might be as simple as adding an antiferromagnetic 2nd neighbour to create a helix - like in [tutorial 3](https://spinw.org/tutorials/03tutorial)... Maybe you can have a go with this? I'll see if I can make some time to look but I'm a bit busy at the moment...

Reply:
I think I already do this in the `test_cmplxBase_equivalent_with_tri` test? Both the tri example and afm example you just suggested end up with `e1 = [0 1 0]` `e2 = [0 0 1]` and `e3 = [1 0 0]` and it's already tested the tri example gives the same results, so I'll leave the cmplxBase tests as they are...

Reply:
I think this is probably good enough for now. I just had a thought that both the `tri` and the chain example systems have only one atom in the structural unit cell, in which case it might be that `'cmplxBase', true` and `'cmplxBase', false` may _always_ give the same `e1`, `e2`, `e3`... It might be that the differences only come up for more complex systems with more than one atom in the unit cell... but I think we can check this later...